 Advances in high-throughput technologies have enabled the genomewide interrogation of genetic variation, the transcriptome, and the epigenome. Translating these advances into personalized health care requires the ability to predict disease outcomes based on idividual omic profiles. In parallel, extensive research in genomics have produced a vast amount of annotations on the function, structure and regulation of the genome. This explosion of genomic information has made genomics-based predictive modeling extremely appealing but also challenging in terms of integrating multiple types of genomics data.  The purpose of the thesis is to develop modeling techniques that can integrate the diverse genomics information to achieve improved prediction performance. 

In chapter 1, we review existing predictive modeling techniques and introduce the problem of integrating annotations and other kinds of external information. We discuss the challenges presented by the high dimensionality of genomics data and the best current tools based on regularized regression. We present detailed information about ridge,  the lasso, the elastic net, and selected nonconvex regularized regression. Since in addition to  prediction, model interpretation is also of great interest, we discuss feature selection properties provided by sparse regularized regression.

Two predictive models for integrating external information into regularized regression for survival outcomes are proposed in chapters 2 and 3. In chapter 2, we present a regularized hierarchical model that integrates meta-features linearly by modeling the regression coefficients to depend on external information through their means. The method developed in chapter 3 allows for differential penalization of features guided by external information. The method can also be viewed as a hierarchical, but  where the regression coefficients depend on the external information through their variance. 

We thus provide two different alternatives for integrating external data into predictive modeling with survival outcomes. We showed in simulations that with informative external data, prediction performance can improve considerably, and that the quality of feature/meta-feature selection also improves. The proposed models are also applied to two high dimensional gene expression datasets: a study of gene expression signatures for breast cancer survival, and an anti-pd1 immunotherapy predictive biomarker study for melanoma survival. 

Chapter 4 introduces a novel algorithm to solve $L_0$-regularized least squares based on proximal distance algorithm. This extends convex regularization methods like lasso and ridge to nonconvex approaches for sparser model and less biased coefficient estimation. 

Perspectives about data integration, high dimensionality, feature selection, and statistical inference are discussed in chapter 5, where we also propose potential future work.

\bigskip
\textbf{Keywords:} genomics; data integration; regularized regression; feature selection; high dimensional data; convex and nonconvex optimization 